config/logs.js

```diff
@@ -0,0 +1,114 @@
+const _ = require('lodash');
+
+module.exports = {
+  hidFormatter(level, msg, logObject) {
+    // Clone the logged object to avoid mutating important objects (e.g. request)
+    // in case the code continues executing after being logged.
+    const localLogObject = _.clone(logObject || {});
+
+    // Define our meta with some defaults.
+    const metadata = {};
+    metadata.level = level;
+
+    // Log IP if we have it.
+    if (logObject.request && logObject.request.info && logObject.request.info.remoteAddress) {
+      metadata.ip = localLogObject.request.info.remoteAddress;
+    }
+    if (logObject.request && logObject.request.headers && logObject.request.headers['x-forwarded-for']) {
+      metadata.ip = localLogObject.request.headers['x-forwarded-for'];
+    }
+
+    // Include custom user object from log
+    metadata.user = localLogObject.user || {};
```

Comment:
Not sure if the `user` object can contain passwords or other sensitive things that should not be logged? Wouldn't it be safer to sanitize it like done for the `request` below?

Reply:
ooh good point. it often doesn't, but it depends on how the object was loaded. Definitely worth taking a second look and I'll probably end up doing as you recommend and adding some mandatory auto-sanitizing to the log formatter.

Reply:
Actually after looking at the next comment about `oauth` I know this is safe. We never log the entire user object wholesale, so the data inside `user` will normally be the following: id and/or email.